JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/DataUploader.java

```diff
@@ -57,11 +59,39 @@ public static void uploadShips(List<Ship> ships, QueryClient queryClient) throws
         // sets course, speed, location measure IRI in ship objects to be used later
         queryClient.setMeasureIri(ships);
 
-        // add a row in RDB time series data, also updates derivation timestamps
-        queryClient.updateTimeSeriesData(ships);
+        switch (source) {
+            case "JSON":
+
+                LOGGER.info("Upload ship data from JSON.");
+                // add a row in RDB time series data, also updates derivation timestamps
+                queryClient.updateTimeSeriesData(ships);
+
+                break;
+
+            case "RDB":
```

Comment:
A better way to handle the different cases is to use enums

Reply:
This is changed in the latest version to use enums.